### DISCUSSION TITLE
What is the script 'ControlNet-M2M' that installs with ControlNet?

### DISCUSSION THREAD

**Discussion by sraney** [Mikubill/sd-webui-controlnet#431] [Q&A]
After updating my automatic1111 ControlNet last night, I noticed a new script in the dropdown,  'ControlNet-M2M'.  It takes a video and generates frames based on the time count you select below.   Is there a description on how to use it?   I suspect it slipped out and is still is in development.

**Comment by zejacky** [comment]
I also wondered what exactly the function of the script is. I was dragging a Video in the field, writing my prompt and after the rendering was doing infinite pictures without stopping. Looks like kind of fetching pictures/movement etc. from a movie. Was no difference if I was enabling ControlNet. However I have no further idea.

**Reply by jav12z** [reply]
Go to Automatic Settings-Controlnet, an activate Allow other scripts to control this extension

Click Apply settings

Now in text2img, go to scripts a load M2M, load a video, and configure the rest of controlnet as usual but without loading pictures.

Controlnet will proccess the pictures from the video and will create a gif.

**Reply by jimys** [reply]
> Go to Automatic Settings-Controlnet, an activate Allow other scripts to control this extension
> 
> Click Apply settings
> 
> Now in text2img, go to scripts a load M2M, load a video, and configure the rest of controlnet as usual but without loading pictures.
> 
> Controlnet will proccess the pictures from the video and will create a gif.

I chose 10 duration, but seems it never ending, and I can't see anything changing except in the console

**Comment by 4lt3r3go** [comment]
i dont have any ControlNet-M2M in any script dropdown menu, what am i missing here?
i'm going crazy
